.github/workflows/build-pack-release.yml

```diff
@@ -0,0 +1,127 @@
+name: Build-Pack-Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: The version to tag the release with, e.g., 1.2.0, 1.3.0
+        required: true
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+    build-pack-release:
+        runs-on: ${{ matrix.os }}
+
+        env:
+            DOTNET_MULTILEVEL_LOOKUP: 1
+            SNK_BUCKET: srprash-xray-dotnet-assets
+
+        strategy:
+            fail-fast: true
+            matrix:
+                os: [windows-latest]
+
+        steps:
+        - name: Checkout repository
+          uses: actions/checkout@v2
+
+        - name: Configure AWS credentials
+          uses: aws-actions/configure-aws-credentials@v1
+          with:
+            role-to-assume: ${{ secrets.AWS_ASSUME_ROLE_ARN_RELEASE }}
+            aws-region: us-west-2
+
+        - name: Get Strong-Naming Key
+          run: aws s3api get-object --bucket ${{ env.SNK_BUCKET }} --key awsxrayrecorder.dll.snk awsxrayrecorder.dll.snk
+          
+        - name: Install dependencies
+          run: dotnet restore .\sdk\AWSXRayRecorder.sln
+
+        - name: Build solution
+          run: dotnet build .\sdk\AWSXRayRecorder.sln /p:Configuration=Release --no-restore /p:AssemblyOriginatorKeyFile=${{ github.workspace }}\awsxrayrecorder.dll.snk
+
+        - name: Assume signer role
+          run: |
+            $resp = aws sts assume-role --role-arn ${{ secrets.AWS_ARTIFACT_ACCESS_ROLE_ARN }} --external-id ${{ secrets.AWS_ROLE_EXTERNAL_ID }} --role-session-name AWSSigner-Session
+            $ak = echo $resp | jq -r ".Credentials.AccessKeyId"
+            $sk = echo $resp | jq -r ".Credentials.SecretAccessKey"
+            $tk = echo $resp | jq -r ".Credentials.SessionToken"
```

Comment:
Didn't we use an action to assume roles before? Is this more/less secure? How can we be confident these creds will never be leaked?

Reply:
The [`aws-actions/configure-aws-credentials`](https://github.com/aws-actions/configure-aws-credentials#assuming-a-role) action works only if the IAM role's trust policy allows GH OIDC to assume the role. It's not the case here, thus this following this way of assuming role.
I am able to mask the role's temporary credentials but still have concerns over the safety as [it doesn't seem to be reliable](https://github.com/actions/runner/issues/475). 
I'm working with the role owners for a safer way to assume roles in Github.